### PR TITLE
Adding Statefulset to pattern 01

### PR DESCRIPTION
**Pull request by Chathuru**
Adding Statefulset to pattern 01. Issues https://github.com/wso2/kubernetes-apim/issues/210 and https://github.com/wso2/kubernetes-apim/issues/72 fix in pattern 1

### CODE REVIEW

File: advanced/am-pattern-2/templates/am/wso2am-pattern-2-conf-entrypoint.yaml [hunk 1]
```diff
@@ -20,62 +20,62 @@ metadata:
 data:
```
Comment (chirangaalwis) [inline advanced/am-pattern-2/templates/am/wso2am-pattern-2-conf-entrypoint.yaml:20]: @Chathuru a suggestion to improve clarity. This file is for APIM pattern 2. But, this PR does not cover APIM pattern 2 as of now. Shall we integrate this change to the other PR in which you plan on adding APIM pattern 2 changes?

File: advanced/am-pattern-1/templates/am/instance-1/wso2am-pattern-1-am-1-statefullset.yaml [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (Chathuru) [reply advanced/am-pattern-1/templates/am/instance-1/wso2am-pattern-1-am-1-statefullset.yaml]: Fin in commit https://github.com/wso2/kubernetes-apim/pull/346/commits/5b63640f3624a2a1bcf23b3f92523488b87e5aa5

File: advanced/am-pattern-1/templates/am/instance-2/wso2am-pattern-1-am-2-statefullset.yaml [hunk 3]
```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (Chathuru) [reply advanced/am-pattern-1/templates/am/instance-2/wso2am-pattern-1-am-2-statefullset.yaml]: Fin in commit https://github.com/wso2/kubernetes-apim/pull/346/commits/5b63640f3624a2a1bcf23b3f92523488b87e5aa5

File: advanced/am-pattern-2/templates/am/wso2am-pattern-2-conf-entrypoint.yaml (hunk 1 above)
Comment (Chathuru) [reply advanced/am-pattern-2/templates/am/wso2am-pattern-2-conf-entrypoint.yaml]: Fin in commit https://github.com/wso2/kubernetes-apim/pull/346/commits/5b63640f3624a2a1bcf23b3f92523488b87e5aa5

File: advanced/am-pattern-1/templates/am/instance-1/wso2am-pattern-1-am-1-statefullset.yaml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: wso2am-pattern-1-am-1-service
+  namespace : {{ .Release.Namespace }}
+spec:
```
[the remark this answers is not in the document]
Comment (Chathuru) [reply advanced/am-pattern-1/templates/am/instance-1/wso2am-pattern-1-am-1-statefullset.yaml:20]: Fin in commit https://github.com/wso2/kubernetes-apim/pull/346/commits/5b63640f3624a2a1bcf23b3f92523488b87e5aa5. Add Rolling update strategy.

**Pull request merged**